src/store/wallet.js

```diff
@@ -705,33 +712,32 @@ export default {
         session_id: this.state.wallet.sessionId,
         wallet_id: this.state.wallet.walletId,
         label,
-        fee: standardizeWitUnits(parameters.fee, WIT_UNIT.NANO),
+        fee: `${parseInt(standardizeWitUnits(parameters.fee, WIT_UNIT.NANO))}`,
```

Comment:
This converts the priority to an integer, so we lose precision, and any priority lower than 1 will be truncated to 0.

This conversion is needed because the wallet `CreateVttRequest` and similar APIs only accept integer fees (fee has type `u64`). If we knew the weight in advance, we could pass an absolute fee instead of a relative fee, avoiding this problem. Also maybe we could change the wallet API to support fractional priorities. What do you think @aesedepece?

Reply:
The result of each tier fee shown when selecting the miner fee is the priority already multiplied by the tx weight. When testing local, if the priority < 0, the miner fee sent to the wallet to generate the weighted value is 0 as it does not support < 0 nanowit.

Reply:
> This converts the priority to an integer, so we lose precision, and any priority lower than 1 will be truncated to 0.

Yesterday I noticed this issue but couldn't locate what was the line actually performing the troublesome truncation.

> This conversion is needed because the wallet `CreateVttRequest` and similar APIs only accept integer fees (fee has type `u64`). If we knew the weight in advance, we could pass an absolute fee instead of a relative fee, avoiding this problem. Also maybe we could change the wallet API to support fractional priorities. What do you think @aesedepece?

Totally for letting the client know the weight, let them do the `weight` by `priority` multiplicarion on their side, and expect them to provide the absolute fee.

Reply:
> The result of each tier fee shown when selecting the miner fee is the priority already multiplied by the tx weight. When testing local, if the priority < 1, the miner fee sent to the wallet to generate the weighted value is 0 as it does not support < 1 nanowit.

Where did you get the weight from? :thinking: 

With the latest commits on https://github.com/witnet/witnet-rust/pull/2261, the transaction weight is now returned upon calls to `createVtt` and `createDataReq`. Also, fees are now also accepted as string-formatted floats, as priority (aka relative fees) can be less than `1` for reasons.

Reply:
> Where did you get the weight from? thinking

The way it works is that it creates a transaction using the relative fee, and the output of that command has the absolute fee.